### PR TITLE
Bugfix #4042 - An index ptn not matching any index is ok.

### LINKED ISSUE #4042: Elasticsearch compatible search endpoint should return an empty set when no matching index is found
**Is your feature request related to a problem? Please describe.**
Return empty hits arrays if index search patterns do not match any indexes

**Describe the solution you'd like**
* If none of the index patterns match an index. return an empty hits array.
* If some of the patterns match an index, return results for the indexes that are found.

**Describe alternatives you've considered**

**Additional context**
Currently if an index pattern doesn't match any index in quickwit it will fail the entire request. This is difficult to handle when using a CSV of globbed index patterns.  In most cases, we use globs because we are unsure if certain indexes exists or not. In the current state, it will return a 404 where as elasticsearch will return a 200 and an empty hits array

### PR DESCRIPTION
**Pull request by fulmicoton**
Fixes #4042

Tested with rest integration tests (checked against elasticsearch)

### CODE REVIEW

File: quickwit/quickwit-proto/src/codegen/quickwit/quickwit.metastore.rs [hunk 1]
```diff
@@ -20,8 +20,8 @@ pub struct CreateIndexResponse {
 #[allow(clippy::derive_partial_eq_without_eq)]
 #[derive(Clone, PartialEq, ::prost::Message)]
 pub struct ListIndexesMetadataRequest {
-    #[prost(string, tag = "1")]
-    pub query_json: ::prost::alloc::string::String,
+    #[prost(string, repeated, tag = "2")]
+    pub index_ptns: ::prost::alloc::vec::Vec<::prost::alloc::string::String>,
```
Comment (fmassot) [inline quickwit/quickwit-proto/src/codegen/quickwit/quickwit.metastore.rs:24]: I would replace `index_ptns` by `index_id_patterns`

File: quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs [hunk 2]
```diff
@@ -1388,8 +1380,15 @@ fn tags_filter_expression_helper(tags: &TagFilterAst) -> Cond {
...
-fn build_index_id_patterns_sql_query(index_id_patterns: Vec<String>) -> anyhow::Result<String> {
-    assert!(!index_id_patterns.is_empty());
+fn build_index_id_patterns_sql_query(index_id_patterns: &[String]) -> anyhow::Result<String> {
+    if index_id_patterns.is_empty() {
+        anyhow::bail!("The list of index id patterns may not be empty.");
+    }
+    if index_id_patterns == ["*"] {
+        return Ok("SELECT * FROM indexes".to_string());
+    }
+    // TODO Fix me.
```
Comment (fmassot) [inline quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs:1390]: Is the `TODO Fix me.` and the following comment are outdated? 
The check is handled in the `root_search` function and not at the metastore level.

**Review by fmassot — APPROVED** [review]
Can you have a look at the 2 minor comments?

File: quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs (hunk 2 above)
Comment (fulmicoton) [reply quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs:1390]: yes thx

**Pull request merged**